market/contracts.py

```diff
@@ -532,7 +532,13 @@ def add_receipt(self,
         receipt_json["buyer_receipt"]["signature"] = \
             self.keychain.signing_key.sign(receipt, encoder=nacl.encoding.HexEncoder)[:128]
         self.contract["buyer_receipt"] = receipt_json["buyer_receipt"]
-        # TODO: update file system and db
+        self.db.Purchases().update_status(self.contract["buyer_order"]["order"]["ref_hash"], 3)
+        file_path = DATA_FOLDER + "purchases/trade receipts/" + order_id + ".json"
+        with open(file_path, 'w') as outfile:
+            outfile.write(json.dumps(self.contract, indent=4))
+        file_path = DATA_FOLDER + "purchases/in progress/" + order_id + ".json"
+        if os.path.exists(file_path):
```

Comment:
think it might be good to double check the trade receipt .json file has been created correctly before removing from in progress?
(sort of an atomicity check)